README.md

```diff
@@ -122,7 +122,8 @@ determining that location is as follows:
 | budget\_amount | The amount to use for a budget alert | `number` | `null` | no |
 | budget\_monitoring\_notification\_channels | A list of monitoring notification channels in the form `[projects/{project_id}/notificationChannels/{channel_id}]`. A maximum of 5 channels are allowed. | `list(string)` | `[]` | no |
 | credentials\_path | Path to a service account credentials file with rights to run the Project Factory. If this file is absent Terraform will fall back to Application Default Credentials. | `string` | `""` | no |
-| default\_service\_account | Project default service account setting: can be one of `delete`, `deprivilege`, `disable`, or `keep`. | `string` | `"disable"` | no |
+| default\_sa\_restore\_policy | The action to be performed in the default service accounts on the resource destroy. Valid values are NONE and REVERT. If set to REVERT it will attempt to restore all default SAs but in the DEPRIVILEGE action. | `string` | `"REVERT"` | no |
```

Comment:
I don't think we need to make this configurable.

Reply:
it must be... if we DELETE it won't be able to revert after 30 days. So, the user must have to NONE.

Reply:
It's fine for the behavior to be customizable on the provider, with the caveat that if set to "REVERT" I think the failure should not be an error (just a warning). /cc @rileykarson

Reply:
I disagree, I think. Terraform failing to do what you told it to (eg undeleting the deleted SAs) should result in an error. We could consider an additional option like "REVERT_AND_IGNORE_FAILURE" though, I guess.

Reply:
I think we can add another ENUM on_failure IGNORE or FAIL, but it would be an enhancement. I can work on it, but we should implement as-is for now.

Reply:
Normally, I agree. But I'm not sure an error which users can't resolve in any way is particularly helpful.

I'd be fine if we wanted to split it to an additional `REVERT_AND_IGNORE_FAILURE` option instead though.

@thiagonache Can you work on that? For this PR we can stick to `REVERT` and remove the configurability.

Reply:
> Normally, I agree. But I'm not sure an error which users can't resolve in any way is particularly helpful.
> 
> I'd be fine if we wanted to split it to an additional `REVERT_AND_IGNORE_FAILURE` option instead though.
> 
> @thiagonache Can you work on that? For this PR we can stick to `REVERT` and remove the configurability.

I can for sure, but by making it configurable the user can take an action to change to NONE, no?
Also, any strong preference using REVERT_AND_IGNORE_FAILURE vs another ENUM on_failure with values IGNORE or FAIL? I prefer the second one REVERT_AND_IGNORE_FAILURE looks odd, but maybe it's just me being annoying.

Reply:
REVERT_AND_IGNORE_FAILURE imo- "NONE" and "IGNORE" doesn't make much sense to configure.

Reply:
I explicitly don't want this configurable. It's not configurable today and we don't shouldn't change the surface area of the module as part of this change.

I'd prefer `REVERT_AND_IGNORE_FAILURE` as only this specific failure would be ignored (we would *not* ignore failures on original deletion, for example).

Reply:
If it is not going to be configurable, why do we want to have a variable?

Reply:
We don't. I want you to remove the variable.

Reply:
in the provider too? I mean in the provider

Reply:
we don't need to have a restore_policy variable there.

Reply:
There are two different things which need to happen:

1. On the provider, update the `restore_policy` param to accept a `REVERT_AND_IGNORE_FAILURE` option.
2. On this module, remove the variable for restore policy and hardcode to `REVERT_AND_IGNORE_FAILURE`.

Reply:
it does not make sense for me, but okay.

Reply:
you mean it should accept REVERT and REVERT_AND_IGNORE_FAILURE on the provider?

Reply:
yeah... it's clear now.

Reply:
Is it still unclear? Please remove the variable.

Reply:
I thought we are going to merge like that and just remove the variable and the provider handles the error... to me, it does not make sense to remove if it may cause an unrecoverable issue... but it's your call :)

Reply:
We can set to `DELETE` for now then and change to `REVERT_AND_IGNORE_FAILURE` once it's available in the module. I want to avoid introducing a variable we will remove soon after.

Reply:
you mean REVERT, right?
the problem I see is if we delete the service account now and we have no variable the new version **must** be released in less than 30 days.

Reply:
I'm assuming you can work on a provider fix fairly quickly? We could also hold back this release until that's in.

Reply:
which I think is feasible unless something really bad happens.

Reply:
I'll be able to work on it next week and should take no more than one day. So, yeah, it's not a biggy to merge like that... we will have the new version before 30 days for sure

Reply:
Sounds good!

Reply:
https://github.com/hashicorp/terraform-provider-google/pull/7766 I still need to work on tests but we can start to discuss it